topology/graph/indexer.go

```diff
@@ -202,9 +202,12 @@ func NewMetadataIndexer(g *Graph, listenerHandler ListenerHandler, m ElementMatc
 	indexer = &MetadataIndexer{
 		indexes: indexes,
 		Indexer: NewIndexer(g, listenerHandler, func(n *Node) (kv map[string]interface{}) {
+			if len(indexes) == 0 {
```

Comment:
We should forbid the creation of the MetadataIndexer and not silently ignore the error